Odata-docs/webapi-8/tutorials/custom-skiptokenhandler.md

```diff
@@ -0,0 +1,549 @@
+}
+
+public override IQueryable ApplyTo(IQueryable query, SkipTokenQueryOption skipTokenQueryOption, ODataQuerySettings querySettings, ODataQueryOptions queryOptions)
+{
+    var decodedSkipTokenQueryOption = DecodeSkipToken(skipTokenQueryOption);
+    return base.ApplyTo(query, decodedSkipTokenQueryOption, querySettings, queryOptions);
+}
+
+private static SkipTokenQueryOption DecodeSkipToken(SkipTokenQueryOption skipTokenQueryOption)
+{
+    string encodedSkipToken = skipTokenQueryOption.RawValue;
+    string decodedSkipToken = Encoding.UTF8.GetString(Convert.FromBase64String(encodedSkipToken));
+    return new SkipTokenQueryOption(decodedSkipToken, skipTokenQueryOption.Context);
+}
+```
+
+We override the `ApplyTo` methods with our custom logic. We override both the generic `ApplyTo` and generic `ApplyTo<T>` overloads. The methods accept the following parameters:
+
+- `query`: The `IQueryable` or `IQueryable<T>` that represents the query that will fetch the results.
+- `skipTokenQueryOption`: Contains information about the `$skiptoken` query option.
+- `querySettings`: The query settings to use while applying the `$skiptoken` (e.g. the page size).
+- `queryOptions`: Contains information about the other query options in the request.
+
+The `ApplyTo` method should transform the `query` and return a new query that implements the paging logic, i.e. ensures that we only return records from where we left off in the last page, and that we do not exceed the page size. Fortunately, the default skip token handler already takes care of this logic.
+
+Our implementation simply decodes the query option and lets the base class take care of the rest:
+
+```csharp
+var decodedSkipTokenQueryOption = DecodeSkipToken(skipTokenQueryOption);
+return base.ApplyTo(query, decodedSkipTokenQueryOption, querySettings, queryOptions);
+```
+
+To decode the skip token, we created a private helper method `DecodeSkipToken`. This method extracts the value of the `$skiptoken`, decodes it from its base64 encoding then creates a new `SkipTokenQueryOption` instance with the decoded value:
+
+```csharp
+string encodedSkipToken = skipTokenQueryOption.RawValue;
+string decodedSkipToken = Encoding.UTF8.GetString(Convert.FromBase64String(encodedSkipToken));
+return new SkipTokenQueryOption(decodedSkipToken, skipTokenQueryOption.Context);
+```
+
+Now let's try to fetch the second page again:
+
+```http
+GET http://localhost:5000/odata/Customers?$skiptoken=SWQtMg%3d%3d
+```
+
+**Response:**
+
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Customers",
+    "value": [
+        {
+            "Id": 3,
+            "Name": "Customer 3"
+        },
+        {
+            "Id": 4,
+            "Name": "Customer 4"
+        }
+    ],
+    "@odata.nextLink": "http://localhost:5000/odata/Customers?$skiptoken=SWQtNA%3d%3d"
+}
+```
+
+Now we get a successful response!
+
+### Complete `CustomSkipTokenHandler` source code
```

Comment:
I thought it would be valuable to include due to the following reasons:
- It's fairly short
- The tutorial demonstrates builds the class progressively one method at a time
- A reader who's creating the custom handler along with the tutorial might get stuck or miss a step. In that case the "full code" provides a reference that they can cross-check against their own code to see where they might have gone wrong.
- We do not link to any "sample" codebase